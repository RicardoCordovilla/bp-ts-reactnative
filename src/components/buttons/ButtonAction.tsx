import React from 'react'
import { TouchableOpacity, Text, View } from 'react-native'
import styles from './buttonaction.styles'

interface ButtonActionProps {
    title: string
    severity: 'primary' | 'secondary' | 'danger'
    action: () => void
}

const severityStyles = {
    primary: {
        bg: '#ebee40',
        color: '#000',
        border: '#000'
    },
    secondary: {
        bg: '#e1e1eb',
        color: '#a19e9e',
        border: '#a19e9e',
    },
    danger: {
        bg: '#c22424',
        color: '#fff',
        border: 'transparent'
    }
}

const ButtonAction: React.FC<ButtonActionProps> = ({ title, severity, action }) => {
    return (
        <TouchableOpacity onPress={action}>
            <View style={[
                styles.buttonAction_container,
                {
                    backgroundColor: severityStyles[severity].bg,
                    borderColor: severityStyles[severity].border
                }
            ]}>
                <Text
                    style={[styles.buttonAction_title,
                    { color: severityStyles[severity].color }
                    ]}>
                    {title}</Text>
            </View>
        </TouchableOpacity>
    )
}

export default ButtonAction