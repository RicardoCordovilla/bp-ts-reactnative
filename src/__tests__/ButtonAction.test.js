import React from 'react';
import { TouchableOpacity, Text, View } from 'react-native';
import renderer from 'react-test-renderer';
import ButtonAction from './../components/buttons/ButtonAction';

describe('ButtonAction', () => {
    it('renders correctly', () => {
        const mockAction = jest.fn();
        const tree = renderer.create(
            <ButtonAction
                title="Test Button"
                severity="primary"
                action={mockAction}
            />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('calls the action function when pressed', () => {
        const mockAction = jest.fn();
        const component = renderer.create(
            <ButtonAction
                title="Test Button"
                severity="primary"
                action={mockAction}
            />
        );
        const button = component.root.findByType(TouchableOpacity);
        button.props.onPress();
        expect(mockAction).toHaveBeenCalled();
    });
});