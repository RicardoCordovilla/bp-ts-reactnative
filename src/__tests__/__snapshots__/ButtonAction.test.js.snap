// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonAction renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
>
  <View
    style={
      [
        {
          "borderColor": "transparent",
          "borderRadius": 5,
          "borderWidth": 1,
          "marginHorizontal": 10,
          "marginTop": 20,
          "paddingVertical": 20,
          "textAlign": "center",
        },
        {
          "backgroundColor": "#ebee40",
          "borderColor": "#000",
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#346",
            "fontSize": 20,
            "fontWeight": "500",
            "textAlign": "center",
          },
          {
            "color": "#000",
          },
        ]
      }
    >
      Test Button
    </Text>
  </View>
</View>
`;
